src/main/java/duke/SearchEngine.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+/**
+ * Class to encapsulate the search engine
+ */
+public class SearchEngine {
+    private Ui ui;
+
+    private Storage searchStorage;
+
+    /**
+     * Constructor for SearchEngine
+     * @param ui    the ui
+     */
+    public SearchEngine(Ui ui) {
+        this.ui = ui;
+        this.searchStorage = Storage.createStorage("./data/search.txt");
+    }
+
+    /**
+     * Searches the taskList for tasks that contain the input
+     * @param input
+     * @return
```

Comment:
```suggestion
     * @return The TaskList containing all found entries.
```
Should this be left blank?